src/chapter08-porting-wolfHSM.md

```diff
@@ -1 +1,153 @@
+- Client Library: This library can be linked against user applications to facilitate communication with the server
+
+# Ports
+
+## Infineon Aurix TC3XX
+
+(Port in progress)
+The distribution of this  port is restricted by the vendor.  Please contact support@wolfssl.com for access. 
+
+Infineon Aurix TC3xx
+- Up to 6x 300MHz TriCore application cores
+- 1x 100MHz ARM Cortex M3 HSM core
+- Crypto offload: TRNG, AES128, ECDSA, ED25519, SHA
+
+## ST SPC58NN
+
+(Port in progress)
+The distribution of this  port is restricted by the vendor.  Please contact support@wolfssl.com for access. 
+
+ST SPC58NN
+- 3x 200MHz e200z4256 PowerPC application cores
+- 1x 100MHz e200z0 PowerPC HSM core with NVM
+- Crypto offload: TRNG, AES128
+
+## Posix
+
+The Posix port provides multiple and fully functional implementations of different wolfHSM abstractions that can be used to better understand the exact functionality expected for different hardware abstractions.
+
+The Posix port provides:
+- Memory buffer transport
+- TCP transport
+- Unix domain transport
+- NVM device (using a filesystem)
+- Flash device (using a file as a backing store)
+
+## Skeleton 
+
+The Skeleton port source code provides a non-functioning layout to be used as a starting point for future hardware/platform ports.  Each function provides the basic description and expected flow with error cases explained so that ports can be used interchangeably with consistent results.
+
+The Skeleton  port provides implementations of:
+- Transport 
+- NVM Device
+- Flash Device
+- Crytp Device
+
+# WolfHSM Porting Interface
+
+## WolfHSM Porting Interface Explained 
+When porting WolfHSM into a new platform or ours, you will need to implement some hardware-specific interfaces. We can provide helper functions for flash-like devices. You will need to provide (read, erase, program, blank check, etc.). Our library will likely work on top of it. We can then provide the object layer on top of that. You will also need to provide crypto hardware interfaces like TRNG, Keys, and symmetric/asymmetric crypto. From a platform interface perspective, you will have to understand your system's boot sequence and how the application cores get reset or marked to boot off of it at a specific start location. You may also need to control shared memory locations and memory translation units. The configuration you want to select is selected at compile time rather than link time. In your code, you select which libraries you want to use; here's the config of the libraries, and then work that at compile time, which is an AUTOSAR requirement. 
+
+## WolfHSM Porting Interface
+
+Ports must implement hardware-specific interfaces:
+- Non-volatile memory
+- Flash-like devices (read, erase, program, blankcheck, etc)
+
```

Comment:
You might be a little confused here: flash is a type of non-volatile memory. wolfHSM has a generic NVM layer that can be configured to use different NVM providers/implementations. One of those providers (the only one baked into wolfHSM at the time of writing) is an NVM flash implementation, which itself is a generic layer that relies on the underlying flash imlementation to be provided be the user (e.g. the actual hardware-specific driver code that reads/writes/erases for that specific flash chip). So I'd just say "NVM flash" interface is what needs to be implemented by the user.